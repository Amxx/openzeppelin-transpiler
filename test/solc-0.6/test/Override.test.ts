import { shouldTranspileToValidContract } from './LocalContract.behaviour';

describe(`SimpleInheritance contract`, (): void => {
  shouldTranspileToValidContract('Child', {
    Child: {
      path: 'Override',
      fileName: 'Override',
      contracts: ['Child', 'Parent2', 'Parent1'],
    },
  });
});
