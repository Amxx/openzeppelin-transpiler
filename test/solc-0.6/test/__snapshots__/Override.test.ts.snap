// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInheritance contract is converted to a valid ChildUpgradeable contract 1`] = `
"pragma solidity ^0.6.0;
import \\"../Override.sol\\";
import \\"./Initializable.sol\\";

contract Parent1Upgradeable is Initializable {
    function __Parent1_init() internal initializer {
        __Parent1_init_unchained();
    }

    function __Parent1_init_unchained() internal initializer {
        
        
    }

  function foo() public virtual {
  }

  function bar() public virtual {
  }
}

contract Parent2Upgradeable is Initializable {
    function __Parent2_init() internal initializer {
        __Parent2_init_unchained();
    }

    function __Parent2_init_unchained() internal initializer {
        
        
    }

  function foo() public virtual {
  }
}

contract ChildUpgradeable is Initializable, Parent1Upgradeable, Parent2Upgradeable {
    function __Child_init() internal initializer {
        __Parent1_init_unchained();
        __Parent2_init_unchained();
        __Child_init_unchained();
    }

    function __Child_init_unchained() internal initializer {
        
        
    }

  function foo() public override(Parent1Upgradeable, Parent2Upgradeable) {
  }

  function bar() public override {
  }
}
"
`;
